### ISSUE TITLE
Model seem to choose worse performance weight

### ISSUE THREAD

**Issue by sonnguyen129** [dreamquark-ai/tabnet#385]
Here is my code:
```
# scheduler
scheduler_params = dict(
        mode = "max", patience = 5, min_lr = 1e-5, factor = 0.9),                 
scheduler_fn=torch.optim.lr_scheduler.ReduceLROnPlateau

# eval_metric
eval_metric=['f1_weighted']
```

But model seem to choose worse performance weight
```
Training fold 1
Device used : cuda
epoch 0  | loss: 0.25273 | train_f1_weighted: 0.01824 | valid_f1_weighted: 0.01948 |  0:00:15s
epoch 1  | loss: 0.11089 | train_f1_weighted: 0.71188 | valid_f1_weighted: 0.71142 |  0:00:27s
epoch 2  | loss: 0.10465 | train_f1_weighted: 0.91196 | valid_f1_weighted: 0.91018 |  0:00:39s

Early stopping occurred at epoch 2 with best_epoch = 0 and best_valid_f1_weighted = 0.01948
Best weights from best epoch are automatically used!
Successfully saved model at ./fold1.zip
```

Why not choose weight in epoch 2, which has best valid_f1_weighted?

**Comment by Optimox** [comment]
Hello @sonnguyen129,

How did you define your metric 'f1_weighted' ?

Have you set `self._maximize = True` in the __init__ method ?

**Comment by sonnguyen129** [comment]
Here is my metric
```
class F1Weighted(Metric):
    def __init__(self):
        self._name = "f1_weighted"
        self._maximize = False

    def __call__(self, y_true, y_pred):
        return f1_score(y_true,  np.argmax(y_pred, axis = 1), average = 'weighted')
```
I set ```self._maximize = False```

**Comment by Optimox** [comment]
no you should set maximize = True since you want the maximum possible f1 score.

Change that and everything will work as expected.
